erpnext/manufacturing/doctype/production_plan/production_plan.py

```diff
@@ -575,6 +575,15 @@ def set_sub_assembly_items_based_on_level(self, row, bom_data, manufacturing_typ
 
 			self.append("sub_assembly_items", data)
 
+	def check_have_work_orders_completed(self):
+		wo_status = frappe.db.get_list(
+			"Work Order",
+			filters={"production_plan": self.name},
+			fields="status",
+			pluck="status"
+		)
+		return all(map(lambda x: x == "Completed", wo_status))
```

Comment:
```suggestion
		return all(x == "Completed" for x in wo_status))
```

map/lambda unnecessary.